### PR TITLE
Update to use ESLint without formatting rules and plugin with them.

### PR DESCRIPTION
**Pull request by duhrer**
See #17 for details.

### CODE REVIEW

**Comment by duhrer** [comment]
I have cut `eslint-config-fluid@2.1.1-dev.20231104T144458Z.5488e1c.GH-17` to test these changes downstream in `fluid-lint-all`.  This pull should be reviewed concurrently with the upcoming pull against `fluid-lint-all` and merged just before that one so we can cut a full release.

File: index.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ https://github.com/fluid-project/eslint-config-fluid/raw/main/LICENSE.txt
 /* eslint-env node */
 "use strict";
 
+// foo
+
```
Comment (jobara) [inline index.js]: Was this change intentional?

Comment (duhrer) [reply index.js]: Must have been a wrong window kind of thing, thanks for catching it.  It's gone now.

**Comment by duhrer** [comment]
Just a note here, I noticed that the current version was "backward facing", i.e. it matches the previous full release, so dev releases of this work were showing up before the last full release in `npm info eslint-config-fluid versions` (and presumably other places).

I updated this pull to use the next patch release.  I can bump to the next minor release if preferred.

File: package.json [hunk 2]
```diff
@@ -26,11 +26,13 @@
...
+        "eslint-plugin-jsdoc": "39.3.13",
+        "@stylistic/eslint-plugin-js": "1.0.0"
     },
     "devDependencies": {
-        "grunt": "1.5.3",
-        "grunt-eslint": "24.0.0",
+        "grunt": "1.6.1",
+        "grunt-eslint": "24.3.0",
         "grunt-jsonlint": "2.1.3"
     }
```
Comment (greatislander) [inline package.json:37]: This is not strictly related to this PR but I wonder if it might be worth removing `grunt` etc if it's only being used for linting this package and `eslint` is now a dependency? If we were to do so we could change the lint script to `eslint` with an appropriate glob rule to target all the JS and JSON files in this package.

**Review by greatislander — APPROVED** [review]
@duhrer Looks great. I did have one question related to the Grunt dependencies.

File: package.json (hunk 2 above)
Comment (duhrer) [reply package.json:37]: I think that's good to handle in another issue, as this small change is needed to allow any projects to work with recent versions of ESLint.

I wrote that up as #19, and left a comment there for you.  Happy to work on that once we have a chance to talk about what we want.

Comment (duhrer) [reply package.json:37]: I made a first pass at #19 in a separate pull (#20).
